Added new file for respectful code reviews (#354)
DXHeroes/knowledge-base-content: practices/respectful_code_reviews.md

---
slug: respectful-code-review
stages:
  - development
short_description: Respectful code reviews involve giving a constructive feeback.The main goal should be to help deliver a good quality final product.
tags:
  - code review
  - quality product
  - communication
  - conflict
  - feedback
  - knowledge sharing
  - mentor
  - comments
  - code review
keywords:
  - code
  - reviews
  - respect
  - constructive
  - feedback
  - quality
  - comments
  - communication
---
# Respectful Code Reviews

**TL;DR**

Respectful and constructive feedback is the basis of a fruitful collaboration, healthy work environment, and quality product.

## Why You Might Want Respectful Code Reviews

> Code review helps to improve the quality of software projects but negative and unclear code review comments can have unwanted consequences.

Reviewer: ```suggestion
Code review helps you to improve the quality of software projects. However, negative and unclear code review comments can have unwanted consequences.
```

> The right style of communication not only helps to build relationships and healthy work environments but also helps to achieve the final goal of delivering a quality product. It improves the overall health of the system and code.

Reviewer: ```suggestion
The right style of communication helps to build developer relationships and healthy work environments.  Respectful Code Reviews help to achieve the final goal of delivering a quality product. They improve the overall health of the system and code.```

Author: They improve the overall health of the system and code.```
I think extra characters have been added in the suggestion at the end

Author: Also, I meant  by relationships I meant workplace relationships between reviewer and developer(both may or may not be developers). Maybe, we can just keep the following point -The right style of communication helps to build  healthy work environments.  Respectful Code Reviews help to achieve the final goal of delivering a quality product. They improve the overall health of the system and code. 
If this helps clear confusion

Author: Also, by relationships I meant workplace relationships between reviewer and developer(both may or may not be developers). Maybe, we can just keep the following point -The right style of communication helps to build healthy work environments. Respectful Code Reviews help to achieve the final goal of delivering a quality product. They improve the overall health of the system and code.
If this helps clear confusion

Reviewer: I see, use either your first version or the second (fact that I got confused doesn't mean it is not clear for 99 % of people)

Author: ok

> 

Author: added image

> 

Reviewer: Hello, sorry for the late reply. Can you please add it to the article using the following syntax? 

![Alt Name](/files/your image) [Name](https://image source)

> 

Author: I have created the image so there is no image source. I have added the image name

> 

Author: Added to article ![Respectful Code Review](/files/respectful_code_review.jpg).The jpg is also added to files folder

## Problems Respectful Code Reviews Helps to Solve

- [Toxic team culture](/problems/toxic-team-culture)
- Knowledge Stagnation  e.g. open discussions lead to knowledge sharing.
- [Long feedback loops](/problems/long-feedback-loops)
- [Poor code quality](/problems/poor-code-quality)

> 

Reviewer: ```suggestion
- [Finger-Pointing And Blaming](/problems/finger-pointing-and-blaming)
```

I would see the blaming problem as linked. What do you think?

## **How to Implement Respectful Code Reviews**

**Follow defined guidelines**

Define guidelines at the project or company level. Follow style guides and coding standard documents and use them as benchmarks for reviews. These guidelines should be available to all concerned for reference.

**Understand code and functionality before reviewing**.

Have a clear picture of the functionality before you start a review. Read related documentation. You can even ask for a code walkthrough. Clarify doubts, if any.

**Explain review comments**

> Wherever possible, support your comments with an explanation. Suggestions and examples help the developer to understand the comments better. If you are suggesting alternate solutions highlight the advantages.

Reviewer: ```suggestion
Wherever possible, support your comments with an explanation. Suggestions and examples help the developer to understand the comments better. If you are proposing alternative solutions, show the advantages.
```

**Be respectful and clear**

> Use clear and respectful language. The development and review process aims to improve the quality of the product hence make comments keeping the code as the main subject of the discussion.

Reviewer: ```suggestion
Use clear and respectful language. The development and review process aims to improve the quality of the product. Make your code the main subject of the review discussion.
```

**Appreciate good work during code review**

> Recognize good work whenever you see it. Acknowledging good work boosts morale of the developer and lays the foundation for a healthy work environment.

Reviewer: ```suggestion
Recognize good work whenever you see it. Acknowledging good work boosts the morale of the developer and lays the foundation for a healthy work environment.
```

Author: Suggestion applied

 **Try to understand the intent of feedback**

As a developer, you should read the comments keeping in mind that review comments are quality checks for the code. They help to make the code and the product better.

 **Common Pitfalls**

> -  Written communication makes it hard to understand the reasoning behind the comment especially if the developer and reviewer are not acquainted.

Reviewer: ```suggestion
-  Written communication makes it hard to understand the comment's intent, especially if the developer and reviewer don't know each other.
```
